core/commands/keystore.go

```diff
@@ -413,6 +418,105 @@ var keyRmCmd = &cmds.Command{
 	Type: KeyOutputList{},
 }
 
+var keyRotateCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Rotates the ipfs identity.",
+		ShortDescription: `
+Generates a new ipfs identity and saves it to the ipfs config file.
+The daemon must not be running when calling this command.
+
+ipfs uses a repository in the local file system. By default, the repo is
+located at ~/.ipfs. To change the repo location, set the $IPFS_PATH
+environment variable:
+
+    export IPFS_PATH=/path/to/ipfsrepo
+`,
+	},
+	Arguments: []cmds.Argument{},
+	Options: []cmds.Option{
+		cmds.StringOption(oldKeyOptionName, "o", "Keystore name for the old/rotated-out key."),
+		cmds.StringOption(keyStoreTypeOptionName, "t", "type of the key to create: rsa, ed25519").WithDefault(keyStoreAlgorithmDefault),
+		cmds.IntOption(keyStoreSizeOptionName, "s", "size of the key to generate"),
```

Comment:
Not sure what I should do here for the option names. Previously, `type, t` and `size, s` were copied from `ipfs init` and used `algorithm, a` and `bits, b`.  It seems to me like keeping these names consistent with `ipfs key gen` is the most reasonable.

We could also just add more aliases (e.g. `type, t, algorithm, a`), but this seems excessive for a new command. Thoughts?

Reply:
I agree. I'd just do what `ipfs key gen` does and break things (we haven't cut a release yet anyways).